### ISSUE TITLE
Plan is missing from Subscription type definition

### ISSUE THREAD

**Issue by JarrodMFlesch** [stripe/stripe-node#1517]
### Is your feature request related to a problem? Please describe.

The type for `plan` & `quantity` are missing [here](https://github.com/stripe/stripe-node/blob/master/types/2022-08-01/Subscriptions.d.ts#L176) - I am not sure if it is deprecated but it still comes back from `Stripe.subscriptions.list` on the latest api version.

### Describe the solution you'd like

Add the type definition for `plan`, or add a note if deprecated.

### Additional context

I feel like I may be missing something - since the types are also complaining that `quantity` does not exist either. But I am not sure how to type my mock data that I got back directly from the `Stripe.subscriptions.list` function, that includes those two properties.

**Comment by remi-stripe** [comment]
@JarrodMFlesch The `Plan` API is deprecated and has been replaced by the `Price` API in 2020 so we recommend always looking at the Price details including the `price` property. This information is *per item* on your Subscription so you look at `items.data[X].price` and `items.data[X].quantity` for a given item.

The `plan` and `quantity` properties on Subscriptions have been actively deprecated since 2017 when we introduced the ability for a Subscription to have more than one "plan" (or Price now) at a time. When that happened, the top-level property didn't make sense anymore unless you only had one item on your Subscription. Over the years we kept those for compatibility reason but we have undocumented those from our API for a long time already.

This is not missing in the types, it's just considered highly deprecated and we discourage anyone from using it.

This was reported recently too:
* https://github.com/stripe/stripe-node/issues/1512
* https://github.com/stripe/stripe-node/issues/1488
* https://github.com/stripe/stripe-node/issues/1392
